### PR TITLE
fixes #1770 - Refactored rdeckfacts definition to use facts_hash

### PR DESCRIPTION
**Pull request by domcleal**
Replaces GH-752 with tests.

### CODE REVIEW

**Comment by dmitri-d** [comment]
I know this is a bugfix; but perhaps rundeck-associated output should be associated with dedicated routes (and controller methods?). Something like GET host/:id/rundeck?

**Comment by dmitri-d** [comment]
Also, tests ftw.

**Comment by ohadlevy** [comment]
@witlessbird ideally this should be a plugin too ;) but as it stands its broken, so I would first fix it, and second extract it to a plugin....

**Comment by domcleal** [comment]
Yeah, I'm preferring the bug fix first, then find somebody to extract it into a plugin and separate route.  I don't even know what rundeck is, so that won't be me...

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -520,7 +520,10 @@ def set_ip_address
   def rundeck
     rdecktags = puppetclasses_names.map{|k| "class=#{k}"}
     unless self.params["rundeckfacts"].empty?
-      rdecktags += self.params["rundeckfacts"].split(",").map{|rdf| "#{rdf}=#{fact(rdf)[0].value}"}
+      rdecktags += self.params["rundeckfacts"].split(",").map { |rdf|
+        rdf.strip!
```
Comment (ohadlevy) [inline app/models/host/managed.rb]: wont a simple gsub on params makes more sense? e.g.

``` ruby
rdecktags += self.params["rundeckfacts"].gsub('\s+','').split(',').map{ |rdf| "#{rdf}=" + (facts_hash[rdf] || "undefined")}
```

Comment (domcleal) [inline app/models/host/managed.rb]: Splitting on `/\s*,\s*/` feels slightly better to me, one less function call.  But yep, will fix.

Comment (domcleal) [inline app/models/host/managed.rb]: Ah.. doesn't help with trailing spaces.

**Comment by domcleal** [comment]
updated

**Comment by ohadlevy** [comment]
thanks, merged as f8825ac

**Pull request closed without merging**